### PR TITLE
Modal: convert Light DOM LESS to SCSS

### PR DESCRIPTION
**Pull request by BY00565233**
JIRA: SURF-1906

### LGTM's
* [x] Dev LGTM
* [x] Dev LGTM

### CODE REVIEW

File: src/scss/components/modal/_index.scss [hunk 1]
```diff
@@ -1,13 +1,14 @@
-@import (reference) 'vars';
+@import 'vars';
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: Please import `../config` (as that is where `%hxComponent` is defined).

```
@import "../config";
```

File: src/scss/components/modal/_index.scss [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: Double quotes, please.

```
@import "vars";
```

File: src/scss/components/modal/_index.scss [hunk 3]
```diff
@@ -20,7 +21,7 @@ hx-modal {
     flex-shrink: 0;
     line-height: 1;
     order: 1;
-    padding: @space[md];
+    padding: 1.25rem;
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: Please use the `$space` map via `map-get()`

```scss
padding: map-get($space, md);
```

File: src/scss/components/modal/_index.scss [hunk 4]
```diff
@@ -31,25 +32,25 @@ hx-modal {
   }
 
   > hx-div {
-    --hxPaddingLeft: @space[md];
-    --hxPaddingRight: @space[md];
+    --hxPaddingLeft: 1.25rem;
+    --hxPaddingRight: 1.25rem;
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: See comment above about `$space`

File: src/scss/components/modal/_index.scss [hunk 5]
```diff
@@ -31,25 +32,25 @@ hx-modal {
...
     flex-grow: 1;
     order: 2;
 
-    #HXDiv.shadyRoot({
-      padding: 0 @space[md];
-    });
+    @mixin hxDiv-shadyRoot{
+      padding: 0 1.25rem;
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: See comment above about `$space`

File: src/scss/components/modal/_index.scss [hunk 6]
```diff
@@ -31,25 +32,25 @@ hx-modal {
...
+    };
 
     &:first-child {
       margin-top: 4rem;
     }
 
     &:last-child {
-      --hxPaddingBottom: @space[md];
+      --hxPaddingBottom: 1.25rem;
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: See comment above about `$space`

File: src/scss/components/modal/_index.scss [hunk 7]
```diff
@@ -31,25 +32,25 @@ hx-modal {
...
 
-      #HXDiv.shadyRoot({
-        padding-bottom: @space[md];
-      });
+      @mixin hxDiv-shadyRoot{
+        padding-bottom: 1.25rem;
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: See comment above about `$space`

File: src/scss/components/modal/_index.scss [hunk 8]
```diff
@@ -58,15 +59,15 @@ hx-modal {
     flex-shrink: 0;
     justify-content: center;
     order: 3;
-    padding: @space[md];
+    padding: 1.25rem;
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: See comment above about `$space`

File: src/scss/components/modal/_index.scss [hunk 9]
```diff
@@ -58,15 +59,15 @@ hx-modal {
...
 
     .hxBtn + .hxBtn {
-      margin-left: @space[md];
+      margin-left: 1.25rem;
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: See comment above about `$space`

**Comment by CITguy** [comment]
Here's why we have to use `#{...}` with CSS custom properties...

SCSS won't interpolate `$variables` in a `--css-custom-property` definition, so we have to explicitly `#{interpolate}` in SCSS.

CleanShot 2019-10-18 at 10 21 22@2x

File: src/scss/components/modal/_index.scss [hunk 10]
```diff
@@ -31,25 +33,26 @@ hx-modal {
...
+    --hxPaddingLeft: #{map-get($space, md)};
+    --hxPaddingRight: #{map-get($space, md)};
+
     flex-grow: 1;
     order: 2;
 
-    #HXDiv.shadyRoot({
-      padding: 0 @space[md];
-    });
+    @mixin hxDiv-shadyRoot{
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: `hxDiv-shadyRoot` not properly applied

Please correct all similar updates in this file.

* Use `@include`, not `@mixin`.
* Space before opening `{`

```scss
@include hxDiv-shadyRoot {
   ...
}
```

File: src/scss/components/modal/_index.scss [hunk 11]
```diff
@@ -31,25 +33,26 @@ hx-modal {
...
     }
 
     &:last-child {
-      --hxPaddingBottom: @space[md];
+      --hxPaddingBottom: map-get($space, md);
 
-      #HXDiv.shadyRoot({
-        padding-bottom: @space[md];
-      });
+      @mixin hxDiv-shadyRoot{
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: See above comment.

File: src/scss/components/modal/_index.scss (hunk 10 above)
Comment (BY00565233) [reply src/scss/components/modal/_index.scss]: Changes has been done

File: src/scss/components/modal/_index.scss (hunk 11 above)
Comment (BY00565233) [reply src/scss/components/modal/_index.scss]: changes has been done.

File: src/scss/components/modal/_index.scss [hunk 12]
```diff
@@ -31,25 +33,26 @@ hx-modal {
...
+      padding: 0 map-get($space, md);
+    };
 
     &:first-child {
       margin-top: 4rem;
     }
 
     &:last-child {
-      --hxPaddingBottom: @space[md];
+      --hxPaddingBottom: map-get($space, md);
```
Comment (CITguy) [inline src/scss/components/modal/_index.scss]: Need to explicitly interpolate the value...

```scss
--hxPaddingBottom: #{map-get($space, md)};
```

Comment (BY00565233) [reply src/scss/components/modal/_index.scss]: Done the changes.

**Pull request merged**